src/mumble/GlobalShortcut_xdp.cpp

```diff
@@ -0,0 +1,189 @@
+        Q_UNUSED(session_handle);
+        Q_UNUSED(timestamp);
+        Q_UNUSED(options);
+
+        foreach (GlobalShortcut *shortcut, qmShortcuts) {
+            if (shortcut_id == shortcut->objectName()) {
+                shortcut->triggered(true, shortcut_id);
+                shortcut->down(shortcut_id);
+            }
+        }
+    });
+    connect(s_shortcutsInterface, &OrgFreedesktopPortalGlobalShortcutsInterface::Deactivated, this, [this] (const QDBusObjectPath &session_handle, const QString &shortcut_id, qulonglong timestamp, const QVariantMap &options) {
+        Q_UNUSED(session_handle);
+        Q_UNUSED(timestamp);
+        Q_UNUSED(options);
+
+        foreach (GlobalShortcut *shortcut, qmShortcuts) {
+            if (shortcut_id == shortcut->objectName()) {
+                shortcut->triggered(false, shortcut_id);
+            }
+        }
+    });
+    connect(s_shortcutsInterface, &OrgFreedesktopPortalGlobalShortcutsInterface::ShortcutsChanged, this, [this] (const QDBusObjectPath &session_handle, const QList<QPair<QString,QVariantMap>> &shortcuts) {
+        Q_UNUSED(session_handle);
+        if (m_shortcuts != shortcuts) {
+            m_shortcuts = shortcuts;
+            Q_EMIT shortcutsChanged();
+        }
+    });
+
+    QTimer::singleShot(0, this, &GlobalShortcutXdp::createSession);
+}
+
+GlobalShortcutXdp::~GlobalShortcutXdp() {
+
+}
+
+void GlobalShortcutXdp::createSession()
+{
+    XdpShortcuts initialShortcuts;
+    initialShortcuts.reserve(qmShortcuts.count());
+    Global::get().s.qlShortcuts.clear();
+
+    int i = 0;
+    m_ids.resize(qmShortcuts.count());
+    foreach (GlobalShortcut *shortcut, qmShortcuts) {
+        initialShortcuts.append({shortcut->objectName(), {
+            { QStringLiteral("description"), shortcut->name }
+        }});
+
+        Shortcut ssss = { i, {uint(i)}, shortcut->qvDefault, false};
+        m_ids[i] = shortcut->objectName();
+        Global::get().s.qlShortcuts << ssss;
+        ++i;
+    }
+
+    QDBusArgument arg;
+    arg << initialShortcuts;
+    auto reply = s_shortcutsInterface->CreateSession({
+        { QLatin1String("session_handle_token"), "Mumble" },
+        { QLatin1String("handle_token"), QLatin1String("mumble") },
+        { QLatin1String("shortcuts"), QVariant::fromValue(arg) },
+    });
+    reply.waitForFinished();
+    if (reply.isError()) {
+        qWarning() << "Couldn't get reply";
+        qWarning() << "Error:" << reply.error().message();
+    } else {
+        QDBusConnection::sessionBus().connect(QString(),
+                                            reply.value().path(),
+                                            QLatin1String("org.freedesktop.portal.Request"),
+                                            QLatin1String("Response"),
+                                            this,
+                                            SLOT(gotGlobalShortcutsCreateSessionResponse(uint,QVariantMap)));
+    }
+}
+
+void GlobalShortcutXdp::run() {
+	// 🤘🤪
```

Comment:
I suppose this will still be implemented? ^^

Reply:
Good point. So in this case we don't need the separate thread at all. The correct thing would be to have `GlobalShortcutEngine` not be a QThread. We don't need the separate thread here because the event is coming from the portals.

I can remove the ::run() implementation for now, but it's still slightly wasteful, so I'd like to know how you would like to proceed.

Reply:
Hm :thinking: 
I guess the correct thing to do here would indeed be to make `GlobalShortcutEngine` independent of `QThread`. Maybe the most simple version would be to have the following hierarchy:
```cpp
class GlobalShortcutEngine {};
class ThreadedShortcutEngine : public GlobalShortcutEngine, QThread {};
```
and then make all currently existing engines inherit from `ThreadedShortcutEngine` but keep the XDG engine inheriting `GlobalShortcutEngine` directly.
That should solve the issue with minimal effort :eyes:

Reply:
Then simply make `GlobalShortcutEngine` extend `QObject` - that should be enough, shouldn't it?

> If you really want this to be addressed from this PR, we can look into this, but I'm not sure it's all that related.

It's only indirectly related but since the way this is currently implemented represents an inefficiency with the new implementation, I would include it in this PR.
Imo it'd be best to do this in a separate `REFAC` commit that can (later) be ordered to be the first commit in this PR...